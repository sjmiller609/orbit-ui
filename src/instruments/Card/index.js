'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'

const Card = ({ children, footer, header, className }) => {
  return (
    <div className={classnames(s.card, className)}>
      {header && <header>{header}</header>}
      <main>{children}</main>
      {footer && <footer>{footer}</footer>}
    </div>
  )
}

Card.propTypes = {
  header: PropTypes.element,
  children: PropTypes.element.isRequired,
  footer: PropTypes.element,
  className: PropTypes.string,
}

export default Card
