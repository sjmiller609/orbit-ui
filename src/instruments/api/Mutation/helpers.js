const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
