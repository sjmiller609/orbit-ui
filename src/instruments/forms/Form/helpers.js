// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
