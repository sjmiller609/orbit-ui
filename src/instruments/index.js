// NOTE: Any instruments importing other instruments must come AFTER

export { default as Row } from './flex/Row'
export { default as Box } from './flex/Box'

export { default as Card } from './cards/Card'
export { default as Table } from './cards/Table'
export { default as TableRow } from './cards/Table/TableRow'
export { default as CardForm } from './cards/CardForm'

export { default as Module } from './layouts/Module'
export { default as Header } from './layouts/Header'

export { default as Loading } from './animations/Loading'
export { default as LoadingDots } from './animations/LoadingDots'

export { default as Search } from './forms/Search'
export { default as Form } from './forms/Form'
export { default as TextField } from './forms/TextField'
export { default as Password } from './forms/Password'
export { default as Checkbox } from './forms/Checkbox'

export { default as Link } from './links/Link'
export { default as Button } from './links/Button'

export { default as Icon } from './icons/Icon'
export { default as Logo } from './icons/Logo'

export { default as P } from './text/P'
export { default as Mini } from './text/Mini'
export { default as ShowDate } from './text/ShowDate'
export { default as B } from './text/B'

export { default as H1 } from './text/H1'
export { default as H2 } from './text/H2'
export { default as H3 } from './text/H3'
export { default as H4 } from './text/H4'
export { default as H5 } from './text/H5'

export { default as Menu } from './menus/Menu'
export { default as Level1 } from './menus/Level1'

export { default as Snackbar } from './ui/Snackbar'

export { default as ContextUI } from './context/UI/Get'
export { default as ProviderUI } from './context/UI'

export { default as Query } from './api/Query'
export { default as Mutation } from './api/Mutation'
