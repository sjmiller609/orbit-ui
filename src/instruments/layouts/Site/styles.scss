@import 'src/styles/mixins.scss';

.content {
  padding: 0 5% 0;

  @include spacing;
}
