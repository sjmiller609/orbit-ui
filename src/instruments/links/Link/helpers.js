// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
