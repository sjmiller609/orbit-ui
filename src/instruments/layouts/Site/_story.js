import React from 'react'
import { storiesOf } from '@storybook/react'
import { Site } from 'instruments'

storiesOf('Instruments|Layouts', module).add('Site', () => (
  <Site nav={<div>[nav element]</div>}>[children]</Site>
))
