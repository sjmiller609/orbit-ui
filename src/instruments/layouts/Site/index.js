'use strict'
import React from 'react'
import PropTypes from 'prop-types'

import classnames from 'classnames'

import s from './styles.scss'

import { Page } from 'instruments'

// wrapper for external pages
const Site = ({ children, nav, metaTitle, className }) => {
  return (
    <Page className={classnames(s.module, className)} metaTitle={metaTitle}>
      {nav}
      <div className={s.content}>{children}</div>
    </Page>
  )
}

Site.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element])
    .isRequired,
  nav: PropTypes.element,
  metaTitle: PropTypes.string,
  className: PropTypes.string,
}

export default Site
