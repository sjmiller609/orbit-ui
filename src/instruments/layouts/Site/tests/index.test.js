import React from 'react'
import { mountWrap } from 'helpers/tests'
import { Page } from 'instruments'
import Site from '../index'

describe('Site', () => {
  it('should be defined', () => {
    expect(Site).toBeDefined()
    expect(Page).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = mountWrap(
      <Site>
        <p>Children</p>
      </Site>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
