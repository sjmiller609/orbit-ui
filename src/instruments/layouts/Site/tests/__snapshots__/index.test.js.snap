// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site should render correctly 1`] = `
<Site>
  <Page
    className="module"
    metaTitle="Astronomer"
  >
    <div
      className="page module"
    >
      <div
        className="content"
      >
        <p>
          Children
        </p>
      </div>
      <HelmetWrapper
        defer={true}
        encodeSpecialCharacters={true}
      >
        <SideEffect(NullComponent)
          defer={true}
          encodeSpecialCharacters={true}
          title="Astronomer"
          titleAttributes={Object {}}
        >
          <NullComponent
            defer={true}
            encodeSpecialCharacters={true}
            title="Astronomer"
            titleAttributes={Object {}}
          />
        </SideEffect(NullComponent)>
      </HelmetWrapper>
    </div>
  </Page>
</Site>
`;
